app/[pokemonName]/components/pokemon-details.jsx

```diff
@@ -1,31 +1,80 @@
 'use client';
 
-import { useState, useEffect } from 'react';
+import { useEffect, useMemo, useState, useCallback } from 'react';
 import Image from 'next/image';
 import Head from 'next/head';
 import styles from './pokemon.module.css';
 import { formatText } from '@/utils/format-text';
 import { useSelector } from 'react-redux';
-import { getPokemonDetails } from '@/redux/features/pokemon/selector';
+import {
+  getPokemonDetails,
+  getIsLoading,
+} from '@/redux/features/pokemon/selector';
 import { useAppDispatch } from '@/redux/store';
 import { fetchPokemonDataThunk } from '@/redux/features/pokemon/service';
 
 export const PokemonPage = ({ name }) => {
   const dispatch = useAppDispatch();
   const pokemonData = useSelector(getPokemonDetails(name));
+  const isLoading = useSelector(getIsLoading);
   const [selectedSprite, setSelectedSprite] = useState('');
 
+  const spriteKeys = useMemo(
+    () => Object.keys(pokemonData?.sprites ?? {}),
+    [pokemonData?.sprites]
+  );
+
   useEffect(() => {
     dispatch(fetchPokemonDataThunk({ name }));
-  }, []);
+  }, [name, dispatch]);
```

Comment:
Since name won't change you don't need to add it. Dispatch is not required to be added too, it stays the same.